### PR TITLE
F/dbparser pdbload refact

### PR DESCRIPTION
**Pull request by lbudai**
This PR contains only equivalent changes.
Goal of this PR was to eliminate those huge switch-cases.

### CODE REVIEW

**Comment by bazsi** [comment]
I like where this is going, looks a lot better. We might as well eliminate the braces  where single statements are embedded in if/else that would make the code even more concise.

**Comment by bazsi** [comment]
:+1:

File: modules/dbparser/pdb-load.c [hunk 1]
```diff
@@ -93,21 +98,35 @@ typedef struct _PDBProgramPattern
   PDBRule *rule;
 } PDBProgramPattern;
 
+
+static void
+_pdb_state_stack_push(PDBStateStack *self, gint state)
+{
+  g_assert(self->top < PDB_STATE_STACK_MAX_DEPTH - 1);
```
Comment (ihrwein) [inline modules/dbparser/pdb-load.c]: :+1: for the bounds checking with abort

**Comment by ihrwein** [comment]
:+1:

**Pull request merged**